sources/osgShader/Compiler.js

```diff
@@ -877,7 +888,11 @@ Compiler.prototype = {
                 inputs.eyeVector = this.getOrCreateNormalizedPosition();
 
             this.getNode( nodeName ).inputs( inputs ).outputs( {
-                color: lightedOutput
+                color: lightedOutput,
+                lightEyePos: inputs.lightEyepos, // spot and point only
```

Comment:
it should be inputs.lightEyePos instead of inputs.lightEyepos no ? same for eyedir